lib/Doctrine/Migrations/DependencyFactory.php

```diff
@@ -36,17 +41,39 @@ class DependencyFactory
     /** @var object[] */
     private $dependencies = [];
 
-    public function __construct(Configuration $configuration)
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var Connection */
+    private $connection;
+
+    /** @var callable */
+    private $sorter;
+
+    public function __construct(Configuration $configuration, Connection $connection, LoggerInterface $logger)
     {
         $this->configuration = $configuration;
+        $this->logger        = $logger;
+        $this->connection    = $connection;
+    }
+
+    public function getConfiguration() : Configuration
+    {
+        return $this->configuration;
+    }
+
+    public function getConnection() : Connection
```

Comment:
Still needed? For extension?
In the Code I only see $this->connection, same for the hetzet above